126 UI create profile page read only mode

KCarlile opened the pull request:
See #126

cander commented:
Do you want the boxes around user name and email to span that whole "column" of the UI? They're kinda short fields.  It seems like they could be the same width as the first name.

For the categories, you could hardcode them to "Domestic Whiskey" and "Rum" to reflect the reality of the system. (We must have verisimilitude in our hacks!) 

Uber nit-pick, you don't capitalize "name" for the name field labels, but you capitalize "Category" in followed categories.

KCarlile commented:
> Do you want the boxes around user name and email to span that whole "column" of the UI? They're kinda short fields. It seems like they could be the same width as the first name.

I've wrapped those in the correct row/col grid divs to fix that.

> For the categories, you could hardcode them to "Domestic Whiskey" and "Rum" to reflect the reality of the system. (We must have verisimilitude in our hacks!)

I had too look up _verisimilitude_. 😄 I've set those as the two checkbox options, but they aren't hooked up to anything yet, obviously.

> Uber nit-pick, you don't capitalize "name" for the name field labels, but you capitalize "Category" in followed categories.

Thanks. I waffle back and forth on those types of things, but I intend for them to at least be internally consistent. I've fixed that.

----

PR updated with changes.

cander commented:
Non nit-pick, I'm not sure the inputs  you have with the form will play right/easily with Rails.

If you look at the stuff about views in https://guides.rubyonrails.org/getting_started.html#generating-a-controller, you can see some of the built-in "magic" that rails provides. If the input form names match the model fields, it's really simple. I notice that you have `firstName` rather than `first_name`, which is the name of the attribute in the model.
https://guides.rubyonrails.org/form_helpers.html has more info about the helpers/magic in Rails.

Or, we could leave this as-is and fix it up when we go to do the real, editing integration.  I'm happy to work with you on it.

KCarlile commented:
Yeah, I knew I'd have to change form input field names/values, but the formatting should be the same. I was planning to fix all the names once you had the User model and Devise stuff setup and working.

I'd say leave this as-is for now and I can do direct fixes on `main` when you're ready.

cander commented on app/controllers/user_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class UserController < ApplicationController
```
This won't be needed when Devise lands. `current_user` will get us what we want. We'll sort it out when merging.

cander commented on app/views/layouts/content/_user_index.html.erb:
```diff
@@ -0,0 +1,86 @@
+<!-- BEGIN User Index include -->
```
Eventually, this should be `user_show`, at least if it were a view. This is a little different because it's under `layouts/content`. We can sort that out.

KCarlile replied:
Fixing, watch for updated PR.

cander commented on app/views/layouts/content/_user_index.html.erb:
```diff
@@ -0,0 +1,86 @@
+        <div class="row">
+          <div class="col-sm-6 mb-5">
+            <label for="firstName" class="form-label fw-bold">First name</label>
+            <input type="text" class="form-control" id="firstName" placeholder="" value="<%= @user.first_name %>" required>
```
at some point (when we actually want to edit user info), we need to align these ids with the model attributes - e.g., `first_name` (this ain't Java! 😄 )

KCarlile replied:
Fixing, watch for updated PR.

cander commented on app/views/user/index.html.erb:
```diff
@@ -0,0 +1,10 @@
+<div class="row">
```
This will be `show.html.erb` because it's displaying a single resource - one user.

KCarlile replied:
Fixing, watch for updated PR.

cander commented on app/views/layouts/components/_header.html.erb:
```diff
@@ -22,7 +22,9 @@
             <a class="nav-link dropdown-toggle" href="/user" role="button" data-bs-toggle="dropdown" aria-expanded="false">User</a>
             <ul class="dropdown-menu">
-              <li><a class="dropdown-item" href="/user"><i class="fa-solid fa-user me-2" style="width: 20px;"></i>Profile</a></li>
+              <li><a class="dropdown-item" href="/user"><i class="fa-solid fa-user me-2" style="width: 20px;"></i>User Profile</a></li>
```
I assume this user profile show function is all handled in the front-end - no round-trip to the controller required.
In which case, should this be conditional on the user being logged in?

KCarlile replied:
I can hide/show in the nav based on logged-in status, which is what I was planning to do.

KCarlile replied:
I've updated this to branch based on auth/anon status.

KCarlile commented:
@cander, I've updated this PR with some changes.

- I've flipped the `user_index` page to be `user_show` (path: `/user/1`) and I added a static user listing page at `user_index` (path: `/user`).
- I made the top nav User menu hide/show menu items based on logged in status.
  - Later, we'll either need to add an alias for `/user/profile` to `/user/{$logged_in_users_id}` or something like that. Alternately, we could dynamically write the header nav menu item, but then we'd lose out on caching that, so aliasing the page is probably a better option. ...but I don't know Rails, as we've already established. 😄
- I updated the top nav's User Profile link to point to `/user/1` for now and I did the same thing in the footer nav.
